react/ModalState.js

```diff
@@ -146,9 +150,16 @@ class ModalState extends Component {
     }
 
     if (isDifferentGeoCoords(thisAddressCoords, prevAddressCoords)) {
-      getExternalPickupPoints(thisAddressCoords).then(data =>
-        this.setState({ externalPickupPoints: data.items })
-      )
+      getExternalPickupPoints(thisAddressCoords)
+        .then(data =>
+          this.setState({
+            externalPickupPoints: data.items.map(item => ({
+              ...item.pickupPoint,
+              distance: null,
+            })),
+          })
+        )
+        .catch(() => this.setState({ externalPickupPoints }))
```

Comment:
⛏ Since you are doing it twice, I would move it to a specific function, it would avoid you from changing in one place and not the other.